FSharpTests.Directory.Build.targets

```diff
@@ -1,13 +1,14 @@
 <Project>
 
-  <UsingTask TaskName="Fsi" AssemblyFile="$(FSharpBuildAssemblyFile)" />
+  <UsingTask TaskName="FsiTestTask" AssemblyFile="$(FSharpBuildAssemblyFile)" />
 
     <Target
         Name="RunFSharpScript"
         Returns=""
         DependsOnTargets="$(CoreCompileDependsOn)"
     >
-        <Fsi  Condition=" '%(_CoreCompileResourceInputs.WithCulture)' != 'true' "
+      <FsiTestTask
```

Comment:
I think it's fine to leave this in as an undocumented artefact of FSharp.Build.

A small downside I realized is for contributors who do not use the boostrapping process - plain "dotnet test ..." will fail for such contributors to this repository until this FsiTestTask gets actually released as part of the SDK.

Which should not be a showstopper, but it is good to have this in mind.
Maybe with @vzarytovskii  we can think of a solution for test targets to make a "shim" for plain test runs w/o boostrap?



(My view: I thinkl it is acceptable if plain "dotnet test" will fail tests assertions for Fsharp.Suite, build script is to be used then. But a plain "dotnet test" would be great to keep working for ComponentTests where new contributions are coming in. Does UsingTask support a condition to exclude this UsingTask in the case of people doing plain `Condition="'$(BUILDING_USING_DOTNET)' == 'true'"` ?)

Reply:
(as in - plain "dotnet test" when not running the build script before, i.e. not having the proto compiler in /artifacts)

Reply:
I guess the solution would be to move this custom task somewhere to FSharp.Test.Utilities?

Reply:
I think that would not help - the use case for the "Fsharp.Compiler.Service.sln" is to trigger the (BUILDING_USING_DOTNET) flag in props and then enable plain dotnet SDK CLI without pre-building anything. So if the task is moved, people would still have to do a 2-phase setup to build it in the first place.

Since it's usage is localized to FSI tests to Fsharp.Suite, it might work to just make sure regular in memory tests are building just fine without it.

Reply:
I'm not sure what exact scenario to test for. I'm out of the loop here probably :)

Reply:
Something among those lines:

git clean -d -f -x (i.e. getting rid of the build script artifacts)
open FSharp.Compiler.Service.sln (e.g. in VS, but any other IDE should do as well)
run some components tests out of the test explorer

Reply:
I went with your initial suggestion to just add parameters to the Fsi tooltask, to make the stdout capture on-demand.
This seems sensible; to not multiply things, we have enough stuff already to care about.

I think the actual assembly loading and parameters passing for the tooltask happens at runtime, so this will not pose any problems to FSharp.Compiler.Service.sln as it does not use it.

I tested the scenario by doing:
`git clean -xdf`
followed by
`dotnet test .\FSharp.Compiler.Service.sln -f "net9.0"`
to verify FSharp.Compiler.Service.sln builds and tests run as expected.

Testing FSharpSuite requires build script, I think, so at that point this tooltask becomes available and the tests pass.